src/modinfo.c

```diff
@@ -3030,6 +3036,16 @@ sr_modinfo_data_load(struct sr_mod_info_s *mod_info, int read_only, sr_session_c
         }
     }
 
+    /* use cached oper push data if available and usable */
+    if ((mod_info->ds == SR_DS_OPERATIONAL) && (mod_info->ds2 == SR_DS_OPERATIONAL) && sess &&
+            sess->oper_push_cache && !mod_info->data) {
+        /* need to duplicate and not consume it, as sr_apply_oper_changes() will consume oper_push_cache */
+        if ((err_info = sr_lyd_dup(sess->oper_push_cache, NULL, LYD_DUP_RECURSIVE | LYD_DUP_WITH_FLAGS, 1, &mod_info->data))) {
```

Comment:
If you really have use-cases with so much push oper data, this seems suboptimal because lots of data may be unnecessary and duplicated redundantly. With the change mentioned above, it should be better to duplicate the data per-module. You can then also check `state` of each module (if needed, not even sure) to handle empty data trees (instead of just checking `modinfo->data`).